src/braket/jobs/local/local_job_container.py

```diff
@@ -146,6 +154,13 @@ def _start_container(self, image_uri: str) -> str:
                     f"The URL {image_uri} is not available locally and can not be pulled from ECR."
                     " Please pull down the container before proceeding."
                 )
+        elif force_update:
+            try:
+                self._pull_image(image_uri)
```

Comment:
Do we want any sort of retry logic for the force update?

Reply:
This just mirrors the existing pull image logic (with the ability to fall back to the existing image). We might want to add retry logic, but it should affect all the "pull_image" paths, not just here, and it would be a bigger (separate) task. However, I'm not sure that it's needed. The only times I've seen this fail is due to my own errors (mistyping something, or needing to update my credentials, etc.).